src/server/connection.py

```diff
@@ -174,7 +177,7 @@ def add(self, peer: Optional[PeerInfo]) -> bool:
         if peer is None or not peer.port:
             return False
         if peer not in self._peers:
-            self._peers.append(peer)
+            self._peers.append(PeerInfo(peer.host, peer.port))
```

Comment:
? Why? Maybe needs a comment

Reply:
Oh I was trying to debug memory leaks. I think this is a useless change